### PR TITLE
[incubator/zookeeper] Support Zookeeper official (3.5.4) image

### PR DESCRIPTION
**Pull request by donbowman**
The existing chart is using gcr.io/google_samples/k8szk, which is not
maintained.

The Zookeeper project has an 'official' image on DockerHub, based
on Zookeeper 3.5.4(-beta). The format of the environment variables
and scripts is different.

Existing issuse such as [#3020](https://github.com/helm/charts/issues/3020)
went stale.

Add support for the Zookeeper official image, mapping the environment
variables and paths.

Signed-off-by: Don Bowman <db@donbowman.ca>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x ] Chart Version bumped
- [x ] Variables are documented in the README.md

### CODE REVIEW

**Review by fabriziofortino — COMMENTED** [review]
That's great work. Should this point to the latest stable release (3.4.13) since 3.5 is still in beta?

**Comment by donbowman** [comment]
> That's great work. Should this point to the latest stable release (3.4.13) since 3.5 is still in beta?

So I changed the defaults to 3.4.13, but I personally would run the beta instead. 3.4.13 and Kubernetes might not play well, if the number of replicas is scaled, etc.

**Comment by donbowman** [comment]
any comments?

**Comment by donbowman** [comment]
any comments?

**Comment by fabriziofortino** [comment]
we would like to see this merged. @lachie83 any thoughts on that?

**Comment by donbowman** [comment]
@lachie83 is there anything more needed from me on this? I'm concerned that people might be using the current version which points to an unmaintained private image.

**Comment by DimitrijeManic** [comment]
Any updates on this PR?

File: incubator/zookeeper/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: zookeeper
 home: https://zookeeper.apache.org/
-version: 1.3.0
-appVersion: 3.4.10
+version: 2.0.0
+appVersion: 3.5.4-beta
```
Comment (davidkarlsen) [inline incubator/zookeeper/Chart.yaml]:
```suggestion
appVersion: 3.4.13
```

File: incubator/zookeeper/README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ This helm chart provides an implementation of the ZooKeeper [StatefulSet](http:/
 * Kubernetes 1.6+
 * PersistentVolume support on the underlying infrastructure
 * A dynamic provisioner for the PersistentVolumes
-* A familiarity with [Apache ZooKeeper 3.4.x](https://zookeeper.apache.org/doc/current/)
+* A familiarity with [Apache ZooKeeper 3.5.x](https://zookeeper.apache.org/doc/r3.5.4-beta/)
```
Comment (davidkarlsen) [inline incubator/zookeeper/README.md]: Need consistent version here too

File: incubator/zookeeper/Chart.yaml (hunk 1 above)
Comment (davidkarlsen) [reply incubator/zookeeper/Chart.yaml]: And version 3.4.14 seems to be out: https://zookeeper.apache.org/releases.html

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: incubator/zookeeper/README.md (hunk 2 above)
Comment (donbowman) [reply incubator/zookeeper/README.md]: ok i'll make these changes and rebase

**Comment by KealanM** [comment]
With 3.5.5 now officially released and the docker image available it would be great to see this updated to use the latest image and merged.

**Comment by donbowman** [comment]
i'm ready to commit w/ all the changes, just waiting on the 3.5.5 image to be available.
https://hub.docker.com/_/zookeeper

**Comment by duncaan** [comment]
@donbowman - the 3.5.5 image was just pushed to dockerhub.  Thanks for working on this!

**Comment by KealanM** [comment]
Any update on this?

**Comment by stackedsax** [comment]
@donbowman can you push your changes now?  We'd love for this to get finished up.

**Comment by donbowman** [comment]
> @donbowman can you push your changes now? We'd love for this to get finished up.

done. sorry this slipped through the cracks in the gap between zookeeper releasing and them releasing the image.

**Comment by stackedsax** [comment]
Thanks very much!

**Comment by MartinX3** [comment]
Thank you for your work :)

Not sure if I did something wrong, but I get a crashloop with the error
```
+ /config-scripts/run
/bin/bash: /config-scripts/run: Permission denied
```

**Comment by MartinX3** [comment]
@donbowman I made two commits to fix a few bugs.
Splitted them into 2 pullrequest, because maybe I am wrong with my fixes.

https://github.com/donbowman/charts/pull/1
https://github.com/donbowman/charts/pull/2

**Comment by donbowman** [comment]
OK seems in 3.5.5 they made that change. I have incorporated your suggestions and tested in Azure w/ a 3-node zookeeper cluster, it came up correctly.

File: incubator/zookeeper/values.yaml [hunk 3]
```diff
@@ -137,7 +112,7 @@ command:
 #   mountPath: /opt/zookeeper/secret
 
 persistence:
-  enabled: true
+  enabled: false
```
Comment (kppullin) [inline incubator/zookeeper/values.yaml]: I was curious as to what would happen when upgrading a deployment of this chart that was originally configured with persistence enabled, with my main concern being that the existing persistent volumes would be deleted.

Fortunately, k8s rejected the change with this error `Error: UPGRADE FAILED: StatefulSet.apps "zookeeper" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden`, preventing any data loss.

Generally, what's the motivation for changing this from `true` to `false`?

**Comment by KealanM** [comment]
@donbowman Any chance we can get some movement on this?

**Comment by donbowman** [comment]
> @donbowman Any chance we can get some movement on this?

what's needed from me?

File: incubator/zookeeper/values.yaml (hunk 3 above)
Comment (donbowman) [reply incubator/zookeeper/values.yaml]: reverted. IMHO the default is wrong but it was not the intent of this PR to change.

**Comment by agolomoodysaada** [comment]
Thanks for this PR! Can we get this merged and then apply improvements in later versions?

**Comment by donbowman** [comment]
So I'm not sure if anything more is needed from me.
All my commits are pushed, i believe it works. All the comments above are addressed.

**Comment by donbowman** [comment]
I still don't know what, if anything, is needed from me. I think this is done.

**Comment by agolomoodysaada** [comment]
@cpanato

**Comment by kppullin** [comment]
Will this change support upgrading an existing deploy of the current chart using the `gcr.io/google_samples/k8szk` image, config, etc?

**Comment by donbowman** [comment]
> Will this change support upgrading an existing deploy of the current chart using the `gcr.io/google_samples/k8szk` image, config, etc?

I didn't test that, its marked as 2.0, a breaking change.
the current one (the google_samples) doesn't work for upgrading itself, the statefulset pvc's go bad.

**Comment by KealanM** [comment]
@lachie83 As your assigned on this could you please comment? It would be great to get this change merged in.

**Pull request merged**